### ISSUE TITLE
Doesn't work with "z" when sourcing ~/.bashrc again

### ISSUE THREAD

**Issue by mkielar** [rcaloras/bash-preexec#103]
So this is a specific case, where I have:
- https://github.com/rupa/z/blob/master/z.sh installed
- https://github.com/rcaloras/bash-preexec installed

My `~/.bashrc` looks like this:
```
source ~/.installers/prompt.sh  # Custom prompt
source ~/.installers/z.sh
source ~/.bash-preexec.sh 
```

The problem is, that Z is changing the `PROMPT_COMMAND` variable to this:
```
→ echo $PROMPT_COMMAND
ps1_generator (_z --add "$(command pwd -P 2>/dev/null)" 2>/dev/null &); history -a; history -c; history -r
```
while `bash-preexec` requires it to be this:
```
→ echo $PROMPT_COMMAND
__bp_precmd_invoke_cmd; __bp_interactive_mode
```

All works fine the first time, but if you `source ~/.bashrc` again, `z` overwites the `PROMPT_COMMAND`, but `bash-preexec` does not, because of the `__bp_imported` being already set. This breaks entire `bash-preexec` functionality in current shell, and I have to restart it to make it work again.

**Comment by mkielar** [comment]
As a workaround, I've added this:
```
PROMPT_COMMAND="__bp_precmd_invoke_cmd; __bp_interactive_mode"
```
below the lines that source `z` and `bash-preexec` and it seems to fix the issue, but it would be much prettier if `bash-preexec` could check which of it's globals were modified, instead of blindly trusting the `__bp_imported` flag.

**Comment by d630** [comment]
Try to use `_Z_NO_PROMPT_COMMAND` and to set the hook by yourself

**Comment by dimo414** [comment]
Consider sending a PR to Z to detect that bash-preexec is installed and hooking into it if so. See also #39.

**Comment by rcaloras** [comment]
@mkielar thanks for the feedback. Closing as a duplicate of #39

**Comment by rcaloras** [comment]
Examining this more closely as I'm working on a fix for #97 and #39. This is a somewhat different issue where bash-preexec breaks if you mutate `PROMPT_COMMAND` after starting your shell such that `__bp_interactive_mode` is not the last thing invoked in`PROMPT_COMMAND`. There's a potential fix for this in that we can have bash-preexec effectively reset PROMPT_COMMAND even if it's been imported.  

Related to z, I think @d630's suggestion is an appropriate fix for this specific case (i.e. Z). In fixing #97, considering if we can make this more flexible.
